### ISSUE TITLE
Add support for combined @supports properties

### ISSUE THREAD

**Issue by chrish-d** [stylelint/stylelint#5495]
> What is the problem you're trying to solve?

When adding multiple `@supports` properties, stylelint incorrectly expects a closing parenthesis.

![Screenshot 2021-08-23 at 17 14 59](https://user-images.githubusercontent.com/8567692/130481587-3fa8a5a9-e42f-4b6c-9010-01f5380990b4.png)

> What solution would you like to see?

As this is valid CSS, we should provide support for it by not flagging it as a linting error.

To re-create:

```css
.test {
  gap: 1em;

  @supports not (display: flex; gap: 1em) {
    & > * {
      margin-bottom: 1em;

      &:last-of-type {
        margin-bottom: 0;
      }
    }
  }
}
```

**Comment by jeddy3** [comment]
@chrish-d Thanks for the report.

I think that error is from another tool, rather than Stylelint. Perhaps https://github.com/microsoft/typescript-styled-plugin.

Neither Stylelint ([demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AXGBnFACYAdAdllgOYCGADvFgIwwC2A3AQUQALoCu55EATiuiz4I2ABRQAlunIAbUgE8qAMxkwAHgxIUqtOgEocLIlgBkWAHxYAVIcLGidUr2IT8AWgBGIlBDo76TPhG9ibwcphuEEpuKPLkMLb29o7Orp7evlQADIFJWAC+wQVGhfilIAA0IOjy+Cikagjg6OiV4BD4ShLETXh2uCC8HKroA1R99gOkKG5Dqm7Cbhz4ANbCAO5BCFgoQzAVRQMeMhBgKwsQbvTksWM7ewd2RAOQJ7wXbq4AbqQyElBuAAW6juuw4+0O7TodBgdQ+11u2zBEKeWAGsDAcl40wkHU8JzOHygXD+YGmMDcYA4mF8bnIvAg8X4EgwoIekIxWJxeOOp3Oi2JsgkZLQdIZTJQLNG2wA2kUiMjHnkJnk0SBusJeDA7nLUXkXh10DAqZKvhTBaTyeg3OsJChAW5JEolDAtXCfjJwdL5cYALo+0r2X1K4zo41cyU8gn8y7oQF8e2kfAA+mM12xSJm3i8f6spHs1EDJQdGZKUh0CQyeREknC8kLct5qiKyHFuFlitVj4VlquYhuYiw13CtwdyvVlYweTrPhQNngkPPEBKZZgSPuMkyMAfb6-f7zlGTZer9eU37bxbLaSkMAwAFp7E+XgHxdqlf4Ne49x-fAwJwD7FJFhGZFmEfBMCTKAnABSQtU-DoX0hSd5CUbEYUdcMnG5dxFgkOgeH4JMUEQwsQBhSRSFHP8UA4LUG3Qy9Vg2LZmwLI98EbAFiBzAEnD-a1cPwD193zBdIQFDBb2TPs3GkY0JC6MA7XkEj2MuC061FaZPnwhNZmGJt7jE0iBVrEUKSNVQ1z4aVWOMtSrnwjN0AgWjb1U0MQEWdRdkoo0K1eQ0PKXQThJg1zjgs8IHUgaFgNsozD080K9xTCB0DtL83G0vCCJmOZtVEpKQsuCRsxgYhhn-dZATtDByBvQq7OKtVUwlatGLWCBNmCtVLONJ86SNDgoEuTFSBaD5li6nqitfAZ+us95yGG0arlUGE4U65jeoWmArMG2J4impjupYrBdTyFs9USVUBg1Pgmoun0jypGk6HW+h4oGF6sH9G7Az9ebql2GTQJgdYfye66j2WO0Tpm86Yc87SCscm5qyhzwYGLLUdR9KYZHWBRpVfIgVX1EB1FvG58Zuo9eTOEZrVIJQ0HedBG3oilGZWZmBjJpIizKiJf0wO8fvpv7BaXB68dlX6l1Z9nKV8TbiJAX7-ryQGiGDSFUYM7mzyNO4BhOdZXQFg38qNjj0OvW9srZq3tkJ4n5GtDL8GIeYoet0jDfmfzhQgE4cIhrHlddqh3ZJgOGejSkTm9-sPGxJ2EUx1weZxx6za8mAswTzzeeT9KZPTxqFkjnPnfZgvfg90nITLzEK59zwM4pX9Ibrjw8-l2OQCbkm3DoYZJTcf3KlbpP29Trvq8d3PcaeuPPdkvs-ZzkulzLtN8Er7ua773965jtVR83ieZCnmfgYP+Ij87qunZXi-nzdkeibH1Od9-HvNUT9YTH2Xg1J2A816N1-p7IBBo3hAnUCbdeIALau2BoFd4wI1DT1hMQe0Bc4wJngVCdW6MMxYygfnBWN0N4txuhTIWVM1A0w1lQS6qolxdF4KLDAaA5yaylvrKW91iCaiepwrhC1YiqB-DMWKjhkzemEQGIoIjXpq2AjaWqaAP6uAyrAGBzdSFvR8B9Nq6ZqxZ2nv3QekiCY-xMbPRhv0BjUxgLTWhXClYu3eGY2klj+AqRcT4t8IsQL8IlkIrhGi7rqnETQjhisr5+NVnFOoQDKYGNzFvH2ACebRklrEtRqI4lLk5FhU8Hgkz9g-tHL+w9fzF1CZ5Spj4so1M7h-ahQ8r6wIYUedp2F8R8lkvQUO4dT5Rz8cYset97671aRUzCHSoxjJDoFdw9TZnf2vl7beFIH4clWSMsumyw54h6fYguzSMEnPGms9w5zDm2PPo4XBwz1wJWoMDL5nSk6+UrDJC54c5lwOWWqf5eItnTLrg08F1oFkSDeYVP5pzTywp2Q3PZAy8m+yOUs9FjyRlYogavJJao7lf2JRGLKNiqE3O8cw-ZWTbqUw8V45JUtfEq2hVsGWnkeF8PFoIrWgqxESLpmEqYaTFHATZUeHJsB8UFNGWcYpqptZJF1tLVsJ4sqKMor3GZOLh77PHpPFFxzSLvnghuNWflyWf0RaqwlgDIVFgNTCx1slnW9NQdS0hdrTwfIodnMWdwsjAxDVle2FIySm2-ugml+qPyngam6e0GAMBwvPsq1BFrkWouDd69wmbgLAiNF7Z1BbblF1dG6ktnrjzpqyjVOq2LL70NIa4WAdRsJ3AAEzA1UD7e0bgABerpLiMTtLtMiDRw2op+cDciEhKIummLRBNlztnOoRbi5xa67wbqotuuiZLq4BvrS0k9FFz00TovGlBBcU2kPXZu6iO66R8TqFW3N+jwK5lvfc0in7H0-uxD7CkD5ph8D9dXQ95qBkftPV+i9FJoODkiEyeDHN-VMpQ8eyEEGACO4JeDZwiEavNFJkP9Obpau+1qiWkfQ24CjrpqMKN9V2xpjG-6vJtUecjlGePpMcIhyBRGqUNtTSZS4MAw46M7eShdiweyL1hPeaILk3I91rh65qwN8AcDoAPd4qhSCSE7tOhkiLSFmYs42xYQKfz9ns+lBdQSMyJtQe+ltaMGV2Ogcyo8RarXNsFkwo8nL2HPR5eE3hkTRWaryOU5hcsHFJdlSreVmSYlatKUGYGi1BrTFBh4Dgoo36TgENJikdbv5BpbeVhDlWczVdFHBwa-HQMKaPO194nWJDddg3hvrhGwtNPk6Q4b6SPCuHw41l1R745tf2gNBDsUlscSm9e2TPbNsHQQ7AdAUkoJwl28twaoErnqbmpCBbfwaO+pNfC3ZxGNtla20tWxb3oROsOzNuTd7nt-cGmGkLkbtjRoh6d5aq0xrhGtP5t93UwNDchwhlaMARoo4mtaCt-6c01urs12b4PSILbxwTz65D0fJsx4NzytPkcM+0bC4Q9R1wF1GhwSK82cfvCOgmiaAWWekLhgoiXGPLas6XB6cEbhkIzl4ACJnw9AvA2V4SwHUmPvnwY-Q5jiyPW69+Cr17vGgerZN04oT+T3VoshHrgHtupPXNBwMVrlvPT65mNDpyEbDJZBKAQfIIB8hAA)) nor our underlying parser ([demo](https://astexplorer.net/#/gist/1d3c21d93219c0a019040b1624464525/14a7262d1701f9a4a835e429c715c7a2b1c64734)) seem to have a problem with it.

I'll close the issue for now. Feel free to reopen with a reproducible demo, though.

---

Incidentally, it's nice to see the `gap` property being used with flexbox in the wild. It's such an improvement over margins for flex items.
